### PR TITLE
Add api for multi-tfm projects

### LINKED ISSUE #183: Add support for multiple MSBuild target frameworks.
I'm not fully aware of the status of MSBuild supporting multiple target frameworks but it's something we'll need to make our ProjectModel aware of.

/cc @natemcmaster

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves #183

cc @NTaylorMullen @prafullbhosale

### CODE REVIEW

File: src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContext.cs [hunk 1]
```diff
@@ -35,7 +35,19 @@ public string FindProperty(string propertyName)
         public string ProjectName => FindProperty("ProjectName") ?? _name;
         public string Configuration { get; }
 
-        public NuGetFramework TargetFramework => NuGetFramework.Parse(FindProperty("NuGetTargetMoniker"));
+        public NuGetFramework TargetFramework
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContext.cs]: Would probably be easier to just swap this to `IEnumerable<NuGetFramework> TargetFrameworks`. That way consuming libraries can choose which framework to use.

File: src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContextBuilder.cs [hunk 2]
```diff
@@ -111,6 +111,22 @@ public virtual MsBuildProjectContext Build(bool ignoreBuildErrors = false)
             return new MsBuildProjectContext(name, _configuration, projectInstance);
         }
 
+        public IEnumerable<MsBuildProjectContext> BuildAllConfigurations()
```
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContextBuilder.cs]: I'd change this to `MsBuildProjectContext Build(NuGetFramework framework)`.

No need to save the caller a for loop if not needed.

Comment (natemcmaster) [inline src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContextBuilder.cs]: There is a `.WithTargetFramework(NuGetFramework)` API already on the builder.

File: src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContext.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContext.cs]: Ah I see. Still too used to old ProjectContext world from project.json 👍

**Pull request merged**